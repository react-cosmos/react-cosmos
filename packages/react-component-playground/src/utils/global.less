:global {
  * {
    margin: 0;
    padding: 0;
  }

  @font-face {
    font-family: 'FiraCode-Light';
    src: url('../static/FiraCode-Light.woff');
  }
}
