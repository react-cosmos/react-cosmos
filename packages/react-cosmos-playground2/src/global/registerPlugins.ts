/* tslint:disable no-var-requires */
import { resetPlugins, enablePlugin } from 'react-plugin';

// Unregistering plugins first makes this file hot-reloadable because a plugin
// can only be registered once with a given name
resetPlugins();

// TODO: Discover plugins in codebase automatically
require('../plugins/Storage').register();
require('../plugins/Core').register();
require('../plugins/Notifications').register();
require('../plugins/Router').register();
require('../plugins/RendererCore').register();
require('../plugins/RendererHeader').register();
require('../plugins/RendererRemote').register();
require('../plugins/RendererPreview').register();
require('../plugins/Nav').register();
require('../plugins/ContentOverlay').register();
require('../plugins/ControlPanel').register();
require('../plugins/ResponsivePreview').register();
require('../plugins/EditFixtureButton').register();

// TODO: Read list of disabled plugins from user config
enablePlugin('controlPanel', false);
