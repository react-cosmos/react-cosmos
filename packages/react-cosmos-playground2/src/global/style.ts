import { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
  html {
    --quick: 0.32s;
    --font-family: -apple-system, BlinkMacSystemFont, Ubuntu, 'Helvetica Neue', Helvetica, sans-serif;
    --hue-primary: 230;
    --hue-accent: 328;
    --hue-success: 120;
    --hue-warning: 48;
    --hue-error: 358;
    --darkest: hsl(var(--hue-primary), 23%, 11%);
    --primary1: hsl(var(--hue-primary), 52%, 22%);
    --primary2: hsl(var(--hue-primary), 54%, 32%);
    --primary3: hsl(var(--hue-primary), 51%, 43%);
    --primary4: hsl(var(--hue-primary), 65%, 63%);
    --primary5: hsl(var(--hue-primary), 69%, 76%);
    --primary6: hsl(var(--hue-primary), 72%, 90%);
    --primary7: hsl(var(--hue-primary), 88%, 97%);
    --grey1: hsl(var(--hue-primary), 21%, 16%);
    --grey2: hsl(var(--hue-primary), 17%, 30%);
    --grey3: hsl(var(--hue-primary), 12%, 49%);
    --grey4: hsl(var(--hue-primary), 25%, 74%);
    --grey5: hsl(var(--hue-primary), 32%, 85%);
    --grey6: hsl(var(--hue-primary), 25%, 91%);
    --grey7: hsl(var(--hue-primary), 17%, 98%);
    --accent1: hsl(var(--hue-accent), 52%, 22%);
    --accent2: hsl(var(--hue-accent), 54%, 32%);
    --accent3: hsl(var(--hue-accent), 51%, 43%);
    --accent4: hsl(var(--hue-accent), 65%, 63%);
    --accent5: hsl(var(--hue-accent), 69%, 76%);
    --accent6: hsl(var(--hue-accent), 72%, 90%);
    --accent7: hsl(var(--hue-accent), 88%, 97%);
    --success1: hsl(var(--hue-success), 52%, 22%);
    --success2: hsl(var(--hue-success), 54%, 32%);
    --success3: hsl(var(--hue-success), 51%, 43%);
    --success4: hsl(var(--hue-success), 60%, 55%);
    --success5: hsl(var(--hue-success), 69%, 70%);
    --success6: hsl(var(--hue-success), 72%, 90%);
    --success7: hsl(var(--hue-success), 88%, 97%);
    --warning1: hsl(var(--hue-warning), 52%, 22%);
    --warning2: hsl(var(--hue-warning), 58%, 32%);
    --warning3: hsl(var(--hue-warning), 76%, 43%);
    --warning4: hsl(var(--hue-warning), 82%, 63%);
    --warning5: hsl(var(--hue-warning), 80%, 76%);
    --warning6: hsl(var(--hue-warning), 85%, 90%);
    --warning7: hsl(var(--hue-warning), 88%, 97%);
    --error1: hsl(var(--hue-error), 52%, 22%);
    --error2: hsl(var(--hue-error), 54%, 32%);
    --error3: hsl(var(--hue-error), 54%, 45%);
    --error4: hsl(var(--hue-error), 72%, 63%);
    --error5: hsl(var(--hue-error), 74%, 78%);
    --error6: hsl(var(--hue-error), 85%, 90%);
    --error7: hsl(var(--hue-error), 88%, 97%);
    font-family: var(--font-family);
    font-size: 14px;
    -webkit-font-smoothing: antialiased;
  }

  body, p, ul, li, h1, h2, h3, button, input, textarea, select {
    margin: 0;
    padding: 0;
  }

  input {
    font-family: var(--font-family);
  }

  input, button {
    font-size: 14px;
  }

  a {}
`;
