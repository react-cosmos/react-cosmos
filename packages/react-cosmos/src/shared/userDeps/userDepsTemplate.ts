import path from 'path';
import slash from 'slash';
import { Json } from './shared';

type Args = {
  globalImports: string[];
  fixturePaths: string[];
  decoratorPaths: string[];
  rendererConfig: Json;
  rootDir: string;
  relativeToDir: string | null;
};
export function userDepsTemplate({
  globalImports,
  fixturePaths,
  decoratorPaths,
  rendererConfig,
  rootDir,
  relativeToDir,
}: Args) {
  const fixtures = createImportMap(fixturePaths, rootDir, relativeToDir);
  const fixtureKeys = Object.keys(fixtures);

  const decorators = createImportMap(decoratorPaths, rootDir, relativeToDir);
  const decoratorKeys = Object.keys(decorators);

  return `
// This file is automatically generated by Cosmos. Add it to .gitignore and
// only edit if you know what you're doing.

// Keeping global imports here is superior to making them bundle entry points
// because this way they become hot-reloadable.
${globalImports
  .map(p => `import '${resolveImportPath(p, relativeToDir)}';`)
  .join(`\n`)}

${fixtureKeys
  .map((k, i) => `import fixture${i} from '${fixtures[k]}';`)
  .join(`\n`)}

${decoratorKeys
  .map((k, i) => `import decorator${i} from '${decorators[k]}';`)
  .join(`\n`)}

export const rendererConfig = ${JSON.stringify(rendererConfig, null, 2)};

export const fixtures = {
${fixtureKeys.map((k, i) => `  '${k}': fixture${i}`).join(`,\n`)}
};

export const decorators = {
${decoratorKeys.map((k, i) => `  '${k}': decorator${i}`).join(`,\n`)}
};
`.trimLeft();
}

function createImportMap(
  paths: string[],
  rootDir: string,
  relativeToDir: string | null
): Record<string, string> {
  return paths.reduce(
    (acc, p) => ({
      ...acc,
      [slash(path.relative(rootDir, p))]: resolveImportPath(p, relativeToDir),
    }),
    {}
  );
}

function resolveImportPath(filePath: string, relativeToDir: string | null) {
  return relativeToDir
    ? `.${path.sep}${path.relative(relativeToDir, filePath)}`
    : filePath;
}
