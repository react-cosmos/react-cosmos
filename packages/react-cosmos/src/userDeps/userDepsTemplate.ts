import {
  userDepsImportMap,
  userDepsImportPath,
  UserDepsTemplateArgs,
} from './userDepsShared.js';

export function userDepsTemplate({
  globalImports,
  fixturePaths,
  decoratorPaths,
  rendererConfig,
  rootDir,
  relativeToDir,
}: UserDepsTemplateArgs) {
  const fixtures = userDepsImportMap(fixturePaths, rootDir, relativeToDir);
  const fixtureKeys = Object.keys(fixtures);

  const decorators = userDepsImportMap(decoratorPaths, rootDir, relativeToDir);
  const decoratorKeys = Object.keys(decorators);

  return `
// This file is automatically generated by Cosmos. Add it to .gitignore and
// only edit if you know what you're doing.

// Keeping global imports here is superior to making them bundle entry points
// because this way they become hot-reloadable.
${globalImports
  .map(p => `import '${userDepsImportPath(p, relativeToDir)}';`)
  .join(`\n`)}

${fixtureKeys
  .map((k, i) => `import fixture${i} from '${fixtures[k]}';`)
  .join(`\n`)}

${decoratorKeys
  .map((k, i) => `import decorator${i} from '${decorators[k]}';`)
  .join(`\n`)}

export const rendererConfig = ${JSON.stringify(rendererConfig, null, 2)};

const fixtures = {
${fixtureKeys
  .map((k, i) => ` '${k}': { module: { default: fixture${i} } }`)
  .join(`,\n`)}
};

const decorators = {
${decoratorKeys
  .map((k, i) => ` '${k}': { module: { default: decorator${i} } }`)
  .join(`,\n`)}
};

export const moduleWrappers = {
  lazy: false,
  fixtures,
  decorators,
};
`.trimStart();
}
