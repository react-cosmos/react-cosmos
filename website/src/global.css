body {
  font-family: -apple-system, BlinkMacSystemFont, Ubuntu, 'Helvetica Neue',
    'Helvetica', 'Arial', sans-serif;
  font-size: 16px;
  margin: 0;
}

html,
body {
  background: #093556;
}

#root {
  color: #0a2e46;
}
