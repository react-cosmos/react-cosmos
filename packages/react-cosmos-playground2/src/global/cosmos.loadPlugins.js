// @flow

import { loadPlugins } from 'react-plugin';

// Statefulness alert!
import './registerPlugins';

loadPlugins({
  config: {
    core: {
      projectId: 'testProjectId',
      fixturesDir: '__jsxfixtures__'
    },
    rendererCoordinator: {
      webUrl: '/_loader.html',
      enableRemote: false
    }
  }
});
