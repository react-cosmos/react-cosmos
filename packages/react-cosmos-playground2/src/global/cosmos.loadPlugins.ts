import { loadPlugins } from 'react-plugin';

// Statefulness alert!
import './registerPlugins';

loadPlugins({
  config: {
    core: {
      projectId: 'testProjectId',
      fixturesDir: '__jsxfixtures__',
      fixtureFileSuffix: 'jsxfixture',
      devServerOn: false,
      webRendererUrl: '/_loader.html'
    }
  }
});
