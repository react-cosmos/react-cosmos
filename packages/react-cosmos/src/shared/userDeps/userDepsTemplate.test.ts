import { userDepsTemplate } from './userDepsTemplate';

const globalImports = [
  '/Users/ovidiu/cosmos/src/polyfills.ts',
  '/Users/ovidiu/cosmos/src/global.css',
];

const fixturePaths = [
  '/Users/ovidiu/cosmos/src/Counter/Counter.fixture.tsx',
  '/Users/ovidiu/cosmos/src/CounterButton/CounterButton.fixture.tsx',
];

const decoratorPaths = ['/Users/ovidiu/cosmos/src/cosmos.decorator.tsx'];

const rendererConfig = { port: 5002 };

it('should generate user deps module with absolute paths', () => {
  expect(
    userDepsTemplate({
      globalImports,
      fixturePaths,
      decoratorPaths,
      rendererConfig,
      rootDir: '/Users/ovidiu/cosmos/',
      relativeToDir: null,
    })
  ).toMatchInlineSnapshot(`
    "// This file is automatically generated by Cosmos. Add it to .gitignore and
    // only edit if you know what you're doing.

    // Keeping global imports here is superior to making them bundle entry points
    // because this way they become hot-reloadable.
    import '/Users/ovidiu/cosmos/src/polyfills.ts';
    import '/Users/ovidiu/cosmos/src/global.css';

    import fixture0 from '/Users/ovidiu/cosmos/src/Counter/Counter.fixture.tsx';
    import fixture1 from '/Users/ovidiu/cosmos/src/CounterButton/CounterButton.fixture.tsx';

    import decorator0 from '/Users/ovidiu/cosmos/src/cosmos.decorator.tsx';

    export const rendererConfig = {
      \\"port\\": 5002
    };

    export const fixtures = {
      'src/Counter/Counter.fixture.tsx': fixture0,
      'src/CounterButton/CounterButton.fixture.tsx': fixture1
    };

    export const decorators = {
      'src/cosmos.decorator.tsx': decorator0
    };
    "
  `);
});

it('should generate user deps module with relative paths', () => {
  expect(
    userDepsTemplate({
      globalImports,
      fixturePaths,
      decoratorPaths,
      rendererConfig,
      rootDir: '/Users/ovidiu/cosmos',
      relativeToDir: '/Users/ovidiu/cosmos/src',
    })
  ).toMatchInlineSnapshot(`
    "// This file is automatically generated by Cosmos. Add it to .gitignore and
    // only edit if you know what you're doing.

    // Keeping global imports here is superior to making them bundle entry points
    // because this way they become hot-reloadable.
    import './polyfills.ts';
    import './global.css';

    import fixture0 from './Counter/Counter.fixture.tsx';
    import fixture1 from './CounterButton/CounterButton.fixture.tsx';

    import decorator0 from './cosmos.decorator.tsx';

    export const rendererConfig = {
      \\"port\\": 5002
    };

    export const fixtures = {
      'src/Counter/Counter.fixture.tsx': fixture0,
      'src/CounterButton/CounterButton.fixture.tsx': fixture1
    };

    export const decorators = {
      'src/cosmos.decorator.tsx': decorator0
    };
    "
  `);
});
