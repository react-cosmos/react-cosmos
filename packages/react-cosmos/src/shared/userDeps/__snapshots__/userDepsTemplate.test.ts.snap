// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate user deps module with absolute paths 1`] = `
"// This file is automatically generated by Cosmos. Add it to .gitignore and
// only edit if you know what you're doing.

// Keeping global imports here is superior to making them bundle entry points
// because this way they become hot-reloadable.
import '/Users/ovidiu/cosmos/src/polyfills.ts';
import '/Users/ovidiu/cosmos/src/global.css';

import fixture0 from '/Users/ovidiu/cosmos/src/Counter/Counter.fixture.tsx';
import fixture1 from '/Users/ovidiu/cosmos/src/CounterButton/CounterButton.fixture.tsx';

import decorator0 from '/Users/ovidiu/cosmos/src/cosmos.decorator.tsx';

export const rendererConfig = {
  \\"port\\": 5002
};

export const fixtures = {
  'src/Counter/Counter.fixture.tsx': fixture0,
  'src/CounterButton/CounterButton.fixture.tsx': fixture1
};

export const decorators = {
  'src/cosmos.decorator.tsx': decorator0
};
"
`;

exports[`should generate user deps module with relative paths 1`] = `
"// This file is automatically generated by Cosmos. Add it to .gitignore and
// only edit if you know what you're doing.

// Keeping global imports here is superior to making them bundle entry points
// because this way they become hot-reloadable.
import './polyfills.ts';
import './global.css';

import fixture0 from './Counter/Counter.fixture.tsx';
import fixture1 from './CounterButton/CounterButton.fixture.tsx';

import decorator0 from './cosmos.decorator.tsx';

export const rendererConfig = {
  \\"port\\": 5002
};

export const fixtures = {
  'src/Counter/Counter.fixture.tsx': fixture0,
  'src/CounterButton/CounterButton.fixture.tsx': fixture1
};

export const decorators = {
  'src/cosmos.decorator.tsx': decorator0
};
"
`;
